controlplane/kubeadm/controllers/controller_test.go

```diff
@@ -1274,14 +1276,20 @@ func createClusterWithControlPlane() (*clusterv1.Cluster, *controlplanev1.Kubead
 
 func createMachineNodePair(name string, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, ready bool) (*clusterv1.Machine, *corev1.Node) {
 	machine := &clusterv1.Machine{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Machine",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: cluster.Namespace,
-			Name:      name,
-			Labels:    internal.ControlPlaneLabelsForClusterWithHash(cluster.Name, hash.Compute(&kcp.Spec)),
+			ClusterName: cluster.Name,
```

Comment:
This is the Kubernetes metadata "cluster name", which isn't anything we use. The one that's relevant to CAPI is in spec (which you have added below on L1292). Let's remove this - thanks!